bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManager.java

```diff
@@ -268,6 +269,23 @@ public void run() {
                     logger.debug("Changed ThingType of Thing {} to {}. New ThingHandler is {}.",
                             thing.getUID().toString(), thing.getThingTypeUID(), thing.getHandler().toString());
                 }
+
+                private void waitForRunningHandlerRegistrations(ThingUID thingUID) {
+                    for (int i = 0; i < 10 * 10; i++) {
```

Comment:
> To be honest, I don't like the whole "locking" mechanism

IMHO code that looks like <code>for (...) { ...; Thread.sleep(...); ...; }</code> should be changed some time.
But ATM I am more happy about non-failing tests.

Perhaps we can improve this chain some time, but I assume it sound more easy as it is.

Thanks for that fix.